### ISSUE TITLE
event: "order open!"

### ISSUE THREAD

**Issue by ahdinosaur** [root-systems/cobuy#41]
job story: when as an admin i've finished the order pre-requisites and i'm ready for my group to begin an order, i should be able to "open" the order. this should assign everyone in the group a TaskPlan to cast their individual intent, which should then send them an email notification to tell them of their current task.

sub-tasks:

- https://github.com/root-systems/cobuy/issues/161

**Comment by sarah-arrrgh** [comment]
depends on #126

**Comment by iainkirkpatrick** [comment]
todo:

- [x] move hook that creates group Agent to be fired when user registers (i.e. a hook on authentication.register from `dogstack-agents`)
  - [x] and same for supplier Agent when that code is merged in (#142)
- [x] TaskPlan created on registration for the user to 'start their first order' - this means the first order for this group is created (see [this issue](https://github.com/root-systems/cobuy/issues/157))
  - [x] this taskPlan has a sub-task of the pre-reqs (i.e. the first step of starting a first order is to do the pre-requisites for any order)

tldr:

- orders are Tasks
- but a groups first order is a 'higher-order' Task, in that there is a wrapper Task that has the normal Order Task and the pre-reqs task (and the pre-reqs must be done first)

thoughts:

- something also there around a pre-req being setting up a supplier, but we probably also want to allow the user to access the pathway in to setting up another supplier after starting another order. so just making sure this feels good as a discrete Task that can be done at the pre-req stage or at this stage in a normal order
- i still have a feeling that it might be cool to also give the first-time user the option of completing the discrete Task of setting up a supplier without having to 'start an order' - i can imagine scenarios where this might be wanted

**Comment by iainkirkpatrick** [comment]
@ahdinosaur should have asked before you left - what does it mean for an 'order' then to be ready? i.e. the pre-reqs are about having a group and a supplier for an order, what else does an order need?

as i see it:

- a supplier selected as the supplier for this order (however the UI works for selecting)
- is that it? :)

**Comment by ahdinosaur** [comment]
- a consumer (group) selected as the consumer for this order.
- a facilitator for whoever is "facilitating" (admin of) the order (the current user).

**Comment by sarah-arrrgh** [comment]
Order setup info:

- Date order closes
- Admin
- Price lists
- Order frequency
- Packers
- Delivery date/time
- Delivery location
- Delivery received by
- Stocktake person
- Packing date/time
- Collect by
- Payment due date
- Payment info (account details + refs)
- Accounts person
- Freight costs
- Other costs
- Provide packaging?
- Penalty if late collection?
- Frozen/chilled?
- Extra notes?

**Comment by iainkirkpatrick** [comment]
thought: perhaps some of the above are 'optional' parameters of an order? perhaps not?

**Comment by danalexilewis** [comment]
@iainkirkpatrick can you add some todos in here like you did before? At the moment it is unclear what remains on the ticket.

**Comment by danalexilewis** [comment]
Omg doing github on a phone sucks :(
